BitwardenShared/Core/Platform/Services/AuthenticatorSyncService.swift

```diff
@@ -102,9 +98,33 @@ actor DefaultAuthenticatorSyncService: NSObject, AuthenticatorSyncService {
     public func start() async {
         guard !started else { return }
         started = true
-        if await configService.getFeatureFlag(FeatureFlag.enableAuthenticatorSync,
-                                              defaultValue: false) {
-            subscribeToAppState()
+
+        guard await configService.getFeatureFlag(FeatureFlag.enableAuthenticatorSync,
+                                                 defaultValue: false) else {
+            return
+        }
+
+        Task {
+            for await (userId, _) in await self.stateService.syncToAuthenticatorPublisher().values {
+                guard let userId else { continue }
+
+                do {
+                    try await determineSyncForUserId(userId)
+                } catch {
+                    errorReporter.log(error: error)
+                }
+            }
+        }
+        Task {
+            for await vaultStatus in await self.vaultTimeoutService.vaultLockStatusPublisher().values {
+                guard let vaultStatus else { continue }
+
+                do {
+                    try await determineSyncForUserId(vaultStatus.userId)
```

Comment:
🤔 I think `vaultLockStatusPublisher()` only publishes the vault lock status for the active account. Does that cause any problems if an inactive account's vault is locked when it times out?

- Unlock vault for account A
- Switch to account B
- Some time later account A's vault times out and locks
- Account A's vault updates in the background, trigging a sync. Decrypting these ciphers would then fail.

We might need a publisher for all accounts for this?

Reply:
I adjusted this for now like we discussed on Friday - added a guard to check if that particular Vault got locked in the background where we wouldn't get the notification. I think this should handle the situation for now and then I can look at if we need the larger all-accounts publisher (I don't think we will)